File: swipe-client/android/src/main/java/com/pl00t/swipe_client/android/AndroidLauncher.java
package com.pl00t.swipe_client.android;

import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.os.Bundle;

import com.badlogic.gdx.Game;
import com.badlogic.gdx.backends.android.AndroidApplication;
import com.badlogic.gdx.backends.android.AndroidApplicationConfiguration;
import com.gameanalytics.sdk.GameAnalytics;
import com.pl00t.swipe_client.SwipeGame;
import com.pl00t.swipe_client.analytics.AnalyticsInterface;

import java.util.Map;

/** Launches the Android application. */
public class AndroidLauncher extends AndroidApplication implements AnalyticsInterface {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        PackageManager pm = getPackageManager();
        try {
            PackageInfo info = pm.getPackageInfo(getPackageName(), 0);
            GameAnalytics.setEnabledInfoLog(true);
            GameAnalytics.setEnabledVerboseLog(true);
            GameAnalytics.configureBuild(info.versionName);
            GameAnalytics.initializeWithGameKey("7e317788e1340f0d6eaa3e65b4d42cc2", "1b95a324841fd91773967d880b4bc98442c8f242");
        } catch (PackageManager.NameNotFoundException e) {
            throw new RuntimeException(e);
        }



        AndroidApplicationConfiguration configuration = new AndroidApplicationConfiguration();
        configuration.useImmersiveMode = true; // Recommended, but not required.
        configuration.numSamples = 2;
        initialize(new SwipeGame(this), configuration);
    }

    @Override
    public void trackEvent(String event, Map<String, String> data) {
        GameAnalytics.addDesignEventWithEventId(event);
    }

    @Override
    public void trackEvent(String event) {
        GameAnalytics.addDesignEventWithEventId(event);
    }

}

File: swipe-client/android/src/main/java/com/pl00t/swipe_client/android/AndroidLauncher.kt
package com.pl00t.swipe_client.android

import android.os.Bundle
import com.badlogic.gdx.backends.android.AndroidApplication
import com.badlogic.gdx.backends.android.AndroidApplicationConfiguration
import com.pl00t.swipe_client.SwipeGame
import com.pl00t.swipe_client.analytics.AnalyticsInterface
import com.yandex.metrica.YandexMetrica
import com.yandex.metrica.YandexMetricaConfig
import java.util.*

/** Launches the Android application.  */
class AndroidLauncher : AndroidApplication(), AnalyticsInterface {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Creating an extended library configuration.
        val config = YandexMetricaConfig.newConfigBuilder(YMAK)
            .withSessionTimeout(15)
            .withLogs()
            .build()
        // Initializing the AppMetrica SDK.
        YandexMetrica.activate(applicationContext, config)
        // Automatic tracking of user activity.
        YandexMetrica.enableActivityAutoTracking(application)
        val sysDefaultSystemLanguage = Locale.getDefault().language.lowercase(Locale.getDefault())
        val configuration = AndroidApplicationConfiguration()
        configuration.useImmersiveMode = true // Recommended, but not required.
        configuration.numSamples = 2
        initialize(
            SwipeGame(
                sysDefaultSystemLanguage,
                this
            ), configuration
        )
    }

    override fun onPause() {
        super.onPause()
        YandexMetrica.getReporter(applicationContext, YMAK).pauseSession()
    }

    override fun onResume() {
        YandexMetrica.getReporter(applicationContext, YMAK).resumeSession()
        super.onResume()
    }

    override fun trackEvent(event: String, data: Map<String, String>) {
        val dataMapped: MutableMap<String, Any> = HashMap()
        for ((key, value) in data) {
            dataMapped[key] = value
        }
        YandexMetrica.getReporter(applicationContext, YMAK).reportEvent(event, dataMapped)
    }

    override fun trackEvent(event: String) {
        YandexMetrica.getReporter(applicationContext, YMAK).reportEvent(event)
    }

    companion object {
        private const val YMAK = "5c63eabc-0a52-4b2d-ae67-b30aec022e11"
    }
}
